With ghc-8.0.1, RationalExamples.lngSecond has ambiguous type variable.

While not a version we're testing with, I found a build error in the examples with ghc-8.0.1.

```
tested-with: >+
  GHC == 7.10.3,
  GHC == 8.0.2,
  GHC == 8.2.2
```

```
uom-plugin> stack build --stack-yaml=stack-8.0.1.yaml
Building all executables for `uom-plugin-examples' once.
After a successful build of all of them, only specified executables will be rebuilt.
uom-plugin-examples-0.1.0.0: build (exe)
Log files have been written to: /.../uom-plugin/.stack-work/logs/

--  While building custom Setup.hs for package uom-plugin-examples-0.1.0.0 using:
      /Users/pdejoux/.stack/.../Cabal-1.24.0.0
build exe:Examples --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /.../uom-plugin/.stack-work/logs/uom-plugin-examples-0.1.0.0.log

    Preprocessing executable 'Examples' for uom-plugin-examples-0.1.0.0...
    [1 of 2] Compiling RationalExamples ( RationalExamples.hs, .stack-work/.../RationalExamples.o )

    /.../uom-plugin/uom-plugin-examples/RationalExamples.hs:53:13: error:
        • Couldn't match type ‘Data.UnitsOfMeasure.Convert.ToCBU t0’
                         with ‘(Base "m" *: One) /: One’
            arising from a use of ‘convert’
          The type variable ‘t0’ is ambiguous
        • In the expression: convert $ (1 / 60) *: lngMinute
          In an equation for ‘lngSecond’:
              lngSecond = convert $ (1 / 60) *: lngMinute
```

--- a/uom-plugin-examples/RationalExamples.hs
+++ b/uom-plugin-examples/RationalExamples.hs
@@ -50,7 +50,10 @@ lngMinuteInNMRational = convert $ toRational' lngMinute
 
 -- A second of longitude at the equator as a distance.
 lngSecond :: Quantity Double [u| m |]
-lngSecond = convert $ (1 / 60) *: lngMinute
+lngSecond =
+    let k = 1 / 60 :: Quantity Double [u| 1 |]
+        s = k *: lngMinute
+    in convert s
 
 lngSecondInFeet:: Quantity Double [u| ft |]
 lngSecondInFeet = convert lngSecond

--- a/uom-plugin-examples/uom-plugin-examples.cabal
+++ b/uom-plugin-examples/uom-plugin-examples.cabal
@@ -2,7 +2,7 @@
 --
 -- see: https://github.com/sol/hpack
 --
--- hash: e671d1e4e31bd29b0906996836777a26db0e4c37299fe8c68e62465c6ebea8d8
+-- hash: b3e6e3b11681ba3b4f00cb9dc90c66f5e1acdd6a8f9e566a43886537348e266d
 
 name:           uom-plugin-examples
 version:        0.1.0.0

--- a/uom-plugin/src/Data/UnitsOfMeasure/Plugin/NormalForm.hs
+++ b/uom-plugin/src/Data/UnitsOfMeasure/Plugin/NormalForm.hs
@@ -1,4 +1,5 @@
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE TupleSections #-}
 module Data.UnitsOfMeasure.Plugin.NormalForm
   ( Atom(..)
   , BaseUnit
@@ -41,9 +42,8 @@ import Util ( thenCmp )
 
 import qualified Data.Foldable as Foldable
 import qualified Data.Map as Map
-import Data.List ( sortBy )
+import Data.List ( sortOn )
 import Data.Maybe
-import Data.Ord
 
 import TcPluginExtras
 
@@ -147,7 +147,7 @@ isConstant = all isBaseLiteral . Map.keys . _NormUnit
 maybeConstant :: NormUnit -> Maybe [(BaseUnit, Integer)]
 maybeConstant = mapM getBase . Map.toList . _NormUnit
   where
-    getBase (BaseAtom ty, i) = (\ b -> (b, i)) <$> isStrLitTy ty
+    getBase (BaseAtom ty, i) = (, i) <$> isStrLitTy ty
     getBase _                = Nothing
 
 -- | Test whether an atom is a base unit (but not necessarily a
@@ -177,7 +177,7 @@ occurs a = any occursAtom . Map.keys . _NormUnit
 
 -- | View a unit as a list of atoms in order of ascending absolute exponent
 ascending :: NormUnit -> [(Atom, Integer)]
-ascending = sortBy (comparing (abs . snd)) . Map.toList . _NormUnit
+ascending = sortOn (abs . snd) . Map.toList . _NormUnit
 
 -- | Drop a variable from a unit
 leftover :: TyVar -> NormUnit -> NormUnit

--- a/uom-plugin/uom-plugin.cabal
+++ b/uom-plugin/uom-plugin.cabal
@@ -2,7 +2,7 @@
 --
 -- see: https://github.com/sol/hpack
 --
--- hash: d110f2f74fb66155437b924f27283bd9b16991b4f2be45693731488d98f4938c
+-- hash: ea266b1e8d435bfb94ff3077fd618465b3182b21be200843bd50a9991478a517
 
 name:           uom-plugin
 version:        0.3.0.0

